src/gitflow/ui/GitflowOpenTaskPanel.java

```diff
@@ -100,19 +106,34 @@ public void commit() {
         final GitflowBranchUtil.ComboEntry selectedFeatureBaseBranch = (GitflowBranchUtil.ComboEntry) featureBaseBranch.getModel().getSelectedItem();
         final GitflowBranchUtil.ComboEntry selectedHotfixBaseBranch = (GitflowBranchUtil.ComboEntry) hotfixBaseBranch.getModel().getSelectedItem();
 
-        GitflowAction action;
-
         if (startFeatureRadioButton.isSelected()) {
-            action = new StartFeatureAction(myRepo);
-            action.runAction(myProject, selectedFeatureBaseBranch.getBranchName(), featureName.getText());
-            gitflowState.setTaskBranch(currentTask, GitflowConfigUtil.getFeaturePrefix(myProject, myRepo) + featureName.getText());
+            final String branchName = GitflowConfigUtil.getFeaturePrefix(myProject, myRepo) + featureName.getText();
+            runAction(new StartFeatureAction(myRepo), selectedFeatureBaseBranch.getBranchName(), branchName);
         }
         else if (startHotfixRadioButton.isSelected()) {
-            action =  new StartHotfixAction(myRepo);
-            action.runAction(myProject, selectedHotfixBaseBranch.getBranchName(), hotfixName.getText());
-            gitflowState.setTaskBranch(currentTask, GitflowConfigUtil.getHotfixPrefix(myProject, myRepo) + hotfixName.getText());
+            final String branchName = GitflowConfigUtil.getHotfixPrefix(myProject, myRepo) + hotfixName.getText();
+            runAction(new StartHotfixAction(myRepo), selectedHotfixBaseBranch.getBranchName(), branchName);
+        }
+    }
+
+    private void runAction(GitflowAction action, String baseBranchName, final String branchName) {
+        final TaskInfo[] current = myVcsTaskHandler.getCurrentTasks();
+
+        if (myPreviousTask != null && myPreviousTask.getBranches(false).isEmpty()) {
+            TaskManagerImpl.addBranches(myPreviousTask, current, false);
```

Comment:
What if the previous task wasn't associated with this branch? Maybe it wasn't associated with any branch in particular... I think it's dangerous to assume that if a current task doesn't have an associated branch then you auto-attach it to the current branch. 

The very least we could have a confirmation popup that tells the user "You are about to switch to a different task, which would change the current branch. Would you like to associate the current task with the current branch before switching?"